.setup/bin/setup_sample_courses.py

```diff
@@ -443,6 +446,185 @@ def create_gradeable_submission(src, dst):
+    """
+    This function generates customization_sample.json in case it has changed from the provided version in the test suite
+    within the Submitty repository. Ideally this function will be pulled out and made independent, or better yet when
+    the code for the web interface is done, that will become the preferred route and this function can be retired.
+
+    Keeping this function after the web interface would mean we have another place where we need to update code anytime
+    the expected format of customization.json changes.
+
+    Right now the code uses the Gradeable and Component classes, so to avoid code duplication the function lives inside
+    setup_sample_courses.py
+
+    :return:
+    """
+    # Right now we don't use the fill-ins, and setup_sample_courses.py lives in the repository so can't do:
+    # customization_path = os.path.join("__INSTALL__FILLIN__SUBMITTY_INSTALL_DIR__", "test_suite", "rainbowGrades")
+    # course_file = os.path.join("__INSTALL__FILLIN__SUBMITTY_REPOSITORY__", ".setup", "data", "courses", "sample.yml")
+
+    # Reseed to minimize the situations under which customization_sample.json changes
+    m = hashlib.md5()
+    m.update("sample")
+    random.seed(int(m.hexdigest(), 16))
+
+    # This won't work because test_suite may not yet exist, also the .json would then be at risk of
+    # "INSTALL_SUBMITTY.sh clean" destroying the .json file.
+    # customization_path = os.path.join(SUBMITTY_INSTALL_DIR, "test_suite", "rainbowGrades")
+
+    customization_path = os.path.join(SUBMITTY_INSTALL_DIR, ".setup")
+    course_file = os.path.join(SUBMITTY_REPOSITORY, ".setup", "data", "courses", "sample.yml")
+    course_json = load_data_yaml(course_file)
+
+    course_id = course_json['code']
+    print("Generating customization_{}.json".format(course_id))
+
+    gradeables = {}
+    gradeables_json_output = ""
+
+    # Read in sample.yml and create gradeables by syllabus bucket
+    for g in course_json['gradeables']:
+        gradeable = Gradeable(g)
+        if gradeable.syllabus_bucket not in gradeables:
+            gradeables[gradeable.syllabus_bucket] = []
+        gradeables[gradeable.syllabus_bucket].append(gradeable)
+
+    # Randomly generate the impact of each bucket on the overall grade
+    gradeables_percentages = []
+    gradeable_percentage_left = 100 - len(gradeables)
+    for i in range(len(gradeables)):
+        gradeables_percentages.append(random.randint(1, gradeable_percentage_left) + 1)
+        gradeable_percentage_left -= (gradeables_percentages[-1] - 1)
+    if gradeable_percentage_left > 0:
+        gradeables_percentages[-1] += gradeable_percentage_left
+
+    # Compute totals and write out each syllabus bucket in the "gradeables" field of customization.json
+    # TODO: Figure out what to do about "none" bucket items - does Rainbow Grades support this?
+    bucket_no = 0
+    for bucket,g_list in gradeables.items():
+        gradeables_json_output += "    {\n" +\
```

Comment:
Right, forgot about just dumping using json.dump() will adapt everywhere I'm currently writing out the whole mess by hand.